init.lua

```diff
@@ -60,16 +54,75 @@ local function re_run_failed_jobs(full_repo, run_id)
     log.info('Api response body is '..res.body)
 end
 
+
+--- Handle an incoming webhook from GitHub
+-- @see https://docs.github.com/en/developers/webhooks-and-events/webhooks/about-webhooks
 function webhook_handler(req)
-    local job = req:json()
-    local run_id = job.workflow_job.run_id
-    local full_repo = job.repository.full_name
-    if is_job_failed(job) and needs_restart(run_id) then
-        fiber.create(function() re_run_failed_jobs(full_repo, run_id) end)
+    local payload = req:json()
+    if payload.workflow_job == nil then
+        log.error('Empty payload')
+        return { status = 502 }
```

Comment:
It could be a HEAD request from GitHub. Let's respond with 204.

Reply:
Also, HTTP codes of replies should be described in the readme.

Reply:
Fixed, added information at Readme.md